fix(build): resolve kustomize installation failures during cross-compilation

therealdwright opened the pull request:
- Build kustomize in isolated temporary directory to avoid module conflicts
- Ensure binary is placed at /go/bin/kustomize for Docker verification step

Fixes Docker build failures in GoReleaser pipeline when cross-compiling
for different target architectures.

Closes #298

devstein reviewed (commented):
Thank you! You are hired 🙇 

Can you fix the failing tests first?

therealdwright committed:
fix(build): resolve kustomize installation failures during cross-compilation

- Build kustomize in isolated temporary directory to avoid module conflicts
- Ensure binary is placed at /go/bin/kustomize for Docker verification step

Fixes Docker build failures in GoReleaser pipeline when cross-compiling
for different target architectures.

Closes #298

therealdwright commented:
@devstein fixed the test errors, the cleanup is probably overkill as these are just build images and I don't want to introduce sudo as is required to handle the cleanup in github-runners. Tested the go-releaser locally which works.
